### PR TITLE
Update to pac4j v1.7 and ratpack 0.9.16

### PR DESCRIPTION
**Pull request by leleuj**
Work in progress...

### CODE REVIEW

File: src/main/java/org/leleuj/AppHandlerFactory.java [hunk 1]
```diff
@@ -28,49 +34,50 @@ public AppHandlerFactory() {
...
+
+                        .handler("theForm.html", new FormHandler(formClient))
+                        .handler("logout.html", new LogoutHandler())
+                        .handler("index.html", new IndexHandler())
+                        .handler("callback", new Pac4jCallbackHandlerBuilder().build())
+        );
     }
 
-    private class AuthenticatedPageChain extends ChainAction {
+    private class AuthenticatedPageChain implements Action<Chain> {
```
Comment (miremond) [inline src/main/java/org/leleuj/AppHandlerFactory.java]: funny to see that it is more and more similar to Play ;)

**Comment by miremond** [comment]
+1

File: src/main/java/org/leleuj/AppHandlerFactory.java (hunk 1 above)
Comment (leleuj) [inline src/main/java/org/leleuj/AppHandlerFactory.java]: `*-pac4j` libraries give us a great overview between all frameworks: Play vs Ratpack vs Undertow vs Spark for example...

**Comment by leleuj** [comment]
Thanks for your confidence, but it still doesn't work: *Work in progress...*

**Comment by miremond** [comment]
A chance that I didn't merge it ;)

**Pull request merged**